lib/ezutils/classes/ezextension.php

```diff
@@ -547,8 +547,13 @@ public static function getHandlerClass( ezpExtensionOptions $options )
                 // only use reflection if we have params to avoid exception on objects withouth constructor
                 if ( $handlerParams !== null && is_array( $handlerParams ) && count( $handlerParams ) > 0 )
                 {
-                    $reflection = new ReflectionClass( $handler );
-                    $object = $reflection->newInstanceArgs( $handlerParams );
+                    // if the class has no constructor do not pass parameters
+                    if ($reflection->getConstructor() != null)
+                    {
+                        $object = $reflection->newInstanceArgs( $handlerParams );
+                    } else {
+                        $object = $reflection->newInstance();
```

Comment:
This is basically the same as the alternative `$object = new $handler();` bellow, perhaps refactor a bit to avoid two separate if / else ?